com.unity.cinemachine/Samples~/Shared Assets/Scripts/SimplePlayerShoot.cs

```diff
@@ -56,6 +63,7 @@ void Update()
                 var go = Instantiate(BulletPrefab, transform.position + fwd, Quaternion.LookRotation(fwd, transform.up));
                 if (go.TryGetComponent<SimpleBullet>(out var b))
                     b.Fire(fwd, BulletSpeed);
+                ImpulseSource.GenerateImpulseWithForce(ImpulseForce);
```

Comment:
what if ImpulseSource is null?

Reply:
I don't like this.  You should create a fire event and set this up in the inspector, not bake Impulse into the code.

Reply:
Fixed